lib/datadog/tracing/context_provider.rb

```diff
@@ -35,32 +37,34 @@ def context(key = nil)
       end
     end
 
-    # ThreadLocalContext can be used as a tracer global reference to create
-    # a different {Datadog::Tracing::Context} for each thread. In synchronous tracer, this
-    # is required to prevent multiple threads sharing the same {Datadog::Tracing::Context}
+    # FiberLocalContext can be used as a tracer global reference to create
+    # a different {Datadog::Tracing::Context} for each fiber. In synchronous tracer, this
+    # is required to prevent multiple fibers sharing the same {Datadog::Tracing::Context}
     # in different executions.
-    class ThreadLocalContext
-      # ThreadLocalContext can be used as a tracer global reference to create
-      # a different {Datadog::Tracing::Context} for each thread. In synchronous tracer, this
-      # is required to prevent multiple threads sharing the same {Datadog::Tracing::Context}
+    #
+    # @see https://ruby-doc.org/core-3.1.2/Thread.html#method-i-5B-5D Thread attributes are fiber-local
+    class FiberLocalContext
+      # FiberLocalContext can be used as a tracer global reference to create
+      # a different {Datadog::Tracing::Context} for each fiber. In synchronous tracer, this
```

Comment:
Minor: I'm curious about what we mean here by "synchronous tracer"...?

Reply:
Yeah, it reads kind of weird.

I rephrased it to actually document some information that would be useful when reading this class.